tests/integration/ovs_test.py

```diff
@@ -387,6 +387,60 @@ def test_ovs_vlan_access_tag():
     assertlib.assert_absent(PORT1)
 
 
+@pytest.mark.tier1
+@pytest.mark.skipif(
+    nm_major_minor_version() < 1.41,
+    reason="OVS VLAN trunks was not supported in NM",
+)
+def test_ovs_vlan_trunks():
+    bridge = Bridge(BRIDGE1)
+    bridge.add_internal_port(PORT1, ipv4_state={InterfaceIPv4.ENABLED: False})
+    bridge.set_port_option(
+        PORT1,
+        {
+            OVSBridge.Port.NAME: PORT1,
+            OVSBridge.Port.VLAN_SUBTREE: {
+                OVSBridge.Port.Vlan.MODE: OVSBridge.Port.Vlan.Mode.TRUNK,
+                OVSBridge.Port.Vlan.TRUNK_TAGS: [
+                    {OVSBridge.Port.Vlan.TrunkTags.ID: 10},
+                    {OVSBridge.Port.Vlan.TrunkTags.ID: 20},
+                    {
+                        OVSBridge.Port.Vlan.TrunkTags.ID_RANGE: {
+                            OVSBridge.Port.Vlan.TrunkTags.MIN_RANGE: 30,
+                            OVSBridge.Port.Vlan.TrunkTags.MAX_RANGE: 40,
+                        }
+                    },
+                ],
+            },
+        },
+    )
+
+    desired_state = bridge.state
+    libnmstate.apply(desired_state)
+    assertlib.assert_state_match(desired_state)
+
+    bridge_config_subtree = desired_state[Interface.KEY][0][
+        OVSBridge.CONFIG_SUBTREE
+    ]
+    bridge_port = bridge_config_subtree[OVSBridge.PORT_SUBTREE][0]
+    bridge_port.update(
+        {
+            OVSBridge.Port.VLAN_SUBTREE: {
+                OVSBridge.Port.Vlan.MODE: OVSBridge.Port.Vlan.Mode.TRUNK
+            }
+        }
+    )
+    libnmstate.apply(desired_state)
+    assertlib.assert_state_match(desired_state)
```

Comment:
I think this was not going to work. The trunks must be set as empty because Nmstate supports partial editing.

Reply:
In addition the test should query the current state see `show_only()` helper and assert explicitly that the trunks are empty.